estate/models/estate_property_offer.py

```diff
@@ -0,0 +1,42 @@
+from odoo import api, fields, models
+from datetime import timedelta
+from odoo.exceptions import UserError
+
+
+class EstatePropertyOffer(models.Model):
+    _name = "estate.property.offer"
+    _description = "Estate Property Offer"
+
+    price = fields.Float(string="Offer Price", required=True)
+    status = fields.Selection([
+        ('accepted', 'Accepted'),
+        ('refused', 'Refused')
+    ], string="Status", copy=False)
+    partner_id = fields.Many2one("res.partner", string="Buyer", required=True)
+    property_id = fields.Many2one("estate.property", string="Property", required=True)
+    
+    validity = fields.Integer(string="Validity (Days)", default=7)
+    date_deadline = fields.Date(string="Deadline", compute="_compute_deadline", inverse="_inverse_deadline", store=True)
+
+    @api.depends("create_date", "validity")
+    def _compute_deadline(self):
+        for record in self:
+            record.date_deadline = record.create_date + timedelta(days=record.validity) if record.create_date else None
+    @api.onchange("date_deadline")        
+    def _inverse_deadline(self):
+        for record in self:
+            if record.date_deadline:
+                record.validity = (record.date_deadline - (record.create_date.date() if record.create_date else fields.Date.today())).days
+                record.validity = max(record.validity,0)
+    def action_accept_offer(self):
+        for record in self:
+            if record.property_id.state == "sold":
+                raise UserError("You cannot accept an offer for a sold property.")
```

Comment:
```suggestion
                raise UserError("You cannot accept an offer for a sold property.")
```
What is the difference b/w `UserError` and `ValidationError`